upstairs/src/volume.rs

```diff
@@ -1536,52 +1558,50 @@ impl Volume {
+                    )?;
+                    sv_results.push(sv_res);
                 }
+                let read_only_parent_compare =
+                    Box::new(match (o_read_only_parent, n_read_only_parent) {
+                        (.., None) => {
+                            // The New VCR is none. When comparing
+                            // read_only_parents this is okay.
+                            CompareResult::NewMissing
+                        }
 
-                Ok(CompareResult::Volume {
-                    sub_compares: vec![Self::compare_vcr_for_replacement(
-                        log,
-                        &o_sub_volumes[0],
-                        &n_sub_volumes[0],
-                        read_only,
-                    )?],
-
-                    read_only_parent_compare: Box::new(
-                        match (o_read_only_parent, n_read_only_parent) {
-                            (.., None) => {
-                                // The New VCR is none. When comparing
-                                // read_only_parents this is okay.
-                                CompareResult::NewMissing
-                            }
+                        (None, Some(..)) => {
+                            // It's never valid when comparing VCRs to have
+                            // the original VCR be missing and the new VCR
+                            // is present.
+                            crucible_bail!(
+                                ReplaceRequestInvalid,
+                                "VCR added where there should not be one"
+                            );
+                        }
 
-                            (None, Some(..)) => {
-                                // It's never valid when comparing VCRs to have
-                                // the original VCR be missing and the new VCR
-                                // is present.
-                                crucible_bail!(
-                                    ReplaceRequestInvalid,
-                                    "VCR added where there should not be one"
-                                );
-                            }
+                        (Some(o_vol), Some(n_vol)) => {
+                            Self::compare_vcr_for_replacement(
+                                log, o_vol, n_vol, true,
+                            )?
+                        }
+                    });
 
-                            (Some(o_vol), Some(n_vol)) => {
-                                Self::compare_vcr_for_replacement(
-                                    log, o_vol, n_vol, true,
-                                )?
-                            }
-                        },
-                    ),
-                })
+                let sv_res = CompareResult::Volume {
+                    sub_compares: sv_results,
+                    read_only_parent_compare,
+                };
+
+                Ok(sv_res.clone())
```

Comment:
It looks like this is the only use of `CompareResult::clone`, and it's unnecessary.  Everything seems to compile fine if I remove it (assuaging my Bad Vibes from above):
```diff
diff --git a/upstairs/src/volume.rs b/upstairs/src/volume.rs
index f61f3f5303..43520a3155 100644
--- a/upstairs/src/volume.rs
+++ b/upstairs/src/volume.rs
@@ -39,7 +39,7 @@
 }

 // Result of comparing two [`VolumeConstructionRequest::Region`] values
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 enum VCRDelta {
     /// The Regions are identical.
     Same,
@@ -50,7 +50,7 @@
 }

 // Result of comparing two [`VolumeConstructionRequest::Volume`] values
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 enum CompareResult {
     Volume {
         sub_compares: Vec<CompareResult>,
@@ -1601,7 +1601,7 @@
                     read_only_parent_compare,
                 };

-                Ok(sv_res.clone())
+                Ok(sv_res)
             }

             (
```

Reply:
Yep, I've made that change, thanks.